airbyte-cdk/python/airbyte_cdk/sources/declarative/declarative_component_schema.yaml

```diff
@@ -1575,6 +1575,18 @@ definitions:
           - FAIL
           - RETRY
           - IGNORE
+      failure_type:
+        title: Failure Type
+        description: Failure type of traced exception if a response matches the filter.
+        type: string
+        enum:
```

Comment:
cross-PR dependencies aren't great, but should we build this on top of @lazebnyi 's work since [he's also adding a rate_limited error type](https://github.com/airbytehq/airbyte/pull/40681)?

Reply:
I'm not sure that it would be a case where we need to override default rate limited behavior and other action except `FAIL`. 
`failure_type` should be used in a combination with action `FAIL`. So in this case [HttpCLient](https://github.com/airbytehq/airbyte/blob/master/airbyte-cdk/python/airbyte_cdk/sources/streams/http/http_client.py#L254) will use failure_type from error_resolution. I added check for used action if failure type provided to response filter and documented it..

Reply:
I think we'll want to modify the HttpClient logic to differentiate between transient and rate limit errors when the action is `RETRY` so we can emit the rate limited messages. 

@lazebnyi, can you help out coordinate and make sure that when this PR and yours are merged in, we'll be able to define which errors should results in rate limit status?

Reply:
To @girarda's point, the RATE_LIMITED ResponseAction PR has been merged and will need to be accounted for here.

Reply:
@pnilan, I don't think I got your point here. Do you want to have ability to raise config error when ResponseAction is RATE_LIMITED? This enum declares available failure types that will be used with FAIL ResponseAction [here](https://github.com/airbytehq/airbyte/blob/master/airbyte-cdk/python/airbyte_cdk/sources/streams/http/http_client.py#L276).

Reply:
@darynaishchenko we should allow the developers to define which errors mean the connector is rate limited.

Concretely, I think this means adding a new `rate_limit_error` type to the [Action enum](https://github.com/airbytehq/airbyte/pull/40676/files#diff-0b89fc73ec03c3a68720d8130993a19002036ff67c8e9e201cb1567fa25507a7R1568) and making sure it maps to the [ResponseAction model used by the error handler](https://github.com/airbytehq/airbyte/blob/master/airbyte-cdk/python/airbyte_cdk/sources/streams/http/error_handlers/response_models.py#L15).

The end result should be that when a response maps to rate limit, the connector emits a rate limit status message to the platform

Reply:
@girarda Thanks, added RATE_LIMITED to Action. So now it will use backoff logic defined for RATE_LIMITED action if it was specified in manifest.